Express like syntax proposal

delvedor opened the pull request:
As titled.

The user has an *express like* API, while we have a more *easy to manage* options object.

mcollina commented on beo.js:
```diff
@@ -8,14 +8,33 @@ function build () {
+      schema: schema,
+      handler: handler
+    })
+  }
```
can you add a unit test for post?

mcollina replied:
we should also add body-parser to read the body (assume JSON for now).

delvedor commented on beo.js:
```diff
@@ -8,14 +8,33 @@ function build () {
   const router = wayfarer()
   const map = new Map()
 
-  beo.route = route
```
Sure!

delvedor commented:
To support POST I've added a body parser, I chose `body` over `body-parser` because body does not change the *req* object.

I also re-exposed `route` and updated the tests.
Finally I've added a little error management .

mcollina commented on beo.js:
```diff
@@ -38,35 +66,49 @@ function build () {
+          handleNode(handle, params, req, res, body)
+        })
+      } else {
+        throw new Error(`${req.method} method is not supported!`)
```
Don't throw. Return a 404.. and add a test about that :D

mcollina commented on beo.js:
```diff
@@ -38,35 +66,49 @@ function build () {
+        jsonParser(req, function (err, body) {
+          if (err) throw err
+          handleNode(handle, params, req, res, body)
+        })
```
this function definition is hoisted, which means it will be allocated even if we do not use it. Move this body-parsing thing into its own function.

mcollina commented on test/shorthand-post.js:
```diff
@@ -0,0 +1,41 @@
+
+beo.post('/', schema, function (req, reply) {
+  reply(null, 200, req.body)
+})
```
wrap this into its own test()
